src/Features/CSharp/Portable/ReplacePropertyWithMethods/CSharpReplacePropertyWithMethodsService.cs

```diff
@@ -305,6 +294,41 @@ public override SyntaxNode GetPropertyNodeToReplace(SyntaxNode propertyDeclarati
             return propertyDeclaration;
         }
 
+        protected override NameMemberCrefSyntax TryGetCrefSyntax(IdentifierNameSyntax identifierName)
+        {
+            return identifierName.Parent as NameMemberCrefSyntax;
+        }
+
+        protected override NameMemberCrefSyntax CreateCrefSyntax(NameMemberCrefSyntax originalCref, SyntaxToken identifierToken, SyntaxNode parameterType)
+        {
+            CrefParameterListSyntax parameterList;
+            if (parameterType is TypeSyntax)
```

Comment:
On second look, the generic method may eliminate the need for the cast... looking again.